src/dispatch/case/flows.py

```diff
@@ -669,6 +669,29 @@ def send_escalation_messages_for_channel_case(
     )
 
 
+def map_case_roles_to_incident_roles(
+    participant_roles: List[ParticipantRoleType], incident: Incident, db_session: Session
+) -> List[ParticipantRoleType]:
+    # Map the case role to an incident role
+    incident_roles = set()
+    for role in participant_roles:
+        if role == ParticipantRoleType.reporter:
+            incident_roles.add(ParticipantRoleType.reporter)
+        elif role == ParticipantRoleType.assignee:
+            # If incident commader role already assigned, assign as participant
+            if participant_service.get_by_incident_id_and_role(
+                db_session=db_session,
+                incident_id=incident.id,
+                role=ParticipantRoleType.incident_commander,
+            ):
+                incident_roles.add(ParticipantRoleType.participant)
+            else:
+                incident_roles.add(ParticipantRoleType.incident_commander)
+        else:
+            incident_roles.add(ParticipantRoleType.participant)
```

Comment:
should we default to observer?

Reply:
I'm not sure. These are folks that were working on the case, so it's less likely they were just acting as observers. Maybe the right way is to add folks as observers and then change to participant once they respond for the first time. I'll write a new ticket for that.

Reply:
> Add folks as observers and then change to participant once they respond for the first time

Definitely sounds like the right way to do it. :+1: